src/pocketmine/scheduler/ClosureTask.php

```diff
@@ -40,20 +40,33 @@ class ClosureTask extends Task{
 
 	/** @var \Closure */
 	private $closure;
+	/** @var \Closure|null */
+	private $cancel;
 
 	/**
 	 * @param \Closure $closure Must accept only ONE parameter, $currentTick
+	 * @param \Closure|null $cancel Will be called if the first argument will return false
 	 */
-	public function __construct(\Closure $closure){
-		Utils::validateCallableSignature(function(int $currentTick) : void{}, $closure);
+	public function __construct(\Closure $closure, ?\Closure $cancel = null){
+		Utils::validateCallableSignature(($nullable = $cancel === null) ? function(int $currentTick) : void{} : function(int $currentTick) : bool{ return true; }, $closure);
+		if(!$nullable){
+			Utils::validateCallableSignature(function() : void{}, $cancel);
+		}
 		$this->closure = $closure;
+		$this->cancel = $cancel;
```

Comment:
I've been thinking about this callback and I don't think it makes a whole lot of sense. A self-cancelling task is capable of running actions directly when it does so.

Reply:
Do you mean that this should be transferred to another class to not to create confusion with the returned type?

Reply:
I mean that this `cancel` closure doesn't make sense.